app/controllers/api/subcollections/flavors.rb

```diff
@@ -0,0 +1,30 @@
+module Api
+  module Subcollections
+    module Flavors
+      def flavors_query_resource(object)
+        object.flavors
+      end
+
+      def flavors_create_resource(parent, _type, _id, data)
+        task_id = Flavor.create_flavor_queue(User.current_user.id, parent, data)
```

Comment:
@imtayadeway @jntullo Should I check 'data' for all needed attributes here? Openstack has it's own check and there will be a check on frontend level.

Reply:
@alexander-demichev no, best to keep that logic out of the controller, and it seems it's already being well handled by Openstack 👍